### ISSUE TITLE
Support for multi-class multi-label problems?

### ISSUE THREAD

**Issue by agamemnonc** [scikit-multilearn/scikit-multilearn#132]
Currently, it seems that multi-class multi-label problems are not supported (e.g. a classification problem with 5 target variables (generally `n_outputs`) where each target variable can take three possible values (generally `n_classes`)). 

Is this something that the authors think that should/should not be supported?

One issue is that currently the output of `predict_proba` methods has `shape = (n_samples, n_outputs)`, whereas to support multi-class multi-label it should be a list with `n_outputs` elements, where each element has `shape = (n_samples, n_classes)` (see output of [`MultioutputClassifier`](http://scikit-learn.org/stable/modules/generated/sklearn.multioutput.MultiOutputClassifier.html#sklearn.multioutput.MultiOutputClassifier.predict_proba) in sklearn).

It seems to me that currently `ClassifierChain` supports multi-class multi-label problems; but `LabelPowerset` doesn't. I have tried to use the `LabelPowerset` transformation to a multi-class multi-label problem that I have, but judging from the output of `unique_combinations_ ` it seems to me that the code currently assumes binary multi-label problems. However, the code runs without any exceptions/warnings, which results in wrong results, therefore I suggest that in case the authors do not want to support multi-class multi-label problems, there should at least be some checks/exceptions/warnings to refrain the user from using such data.

**Comment by ChristianSch** [comment]
> However, the code runs without any exceptions/warnings, which results in wrong results, therefore I suggest that in case the authors do not want to support multi-class multi-label problems, there should at least be some checks/exceptions/warnings to refrain the user from using such data.

In my opinion we shouldn't clutter the code with such things. It's the user's responsibility to check what the code does. We don't indicate that we support multi-target classification anywhere, I guess.
I'm not sure it would be beneficial to the library if we implemented support for this. I didn't look into it, but even now there are few people willing to participate.
@niedakh what's your take?

**Comment by niedakh** [comment]
I think we shouldn't support multi-output tasks, there's no real foundation for that in scipy sparse too, I don't thinkg scipy matrices can handle sparse matrices with additional subvectors. We do rely a lot in the optimizations on this.

**Comment by dyhan316** [comment]
just wanted to ask : does this also apply to `IterativeStratification`? I tried with multi-class multi-label target array, (for instance, arrays like :
```
array([[2, 1, 0],
       [0, 0, 0],
       [1, 1, 0],
       [1, 1, 1],
       [0, 1, 0]])
``` 
(the first label has three possible classes)


After playing around with a bit, it seems that when using `IterativeStratification` with multi class + multi-label does not throw any errors, the resulting stratified folds do not have similar class proportions to the original distributions (i.e. not stratified)
